astropy/time/core.py

```diff
@@ -583,6 +588,40 @@ def value(self):
         # the ``value`` attribute is cached.
         return getattr(self, self.format)
 
+    @property
+    def masked(self):
+        return self._time.masked
+
+    @property
+    def mask(self):
+        return self._time.mask
+
+    def __setitem__(self, item, value):
+        if value in (np.ma.masked, np.nan):
+            self._time.jd2[item] = np.nan
+            del self.cache
+            return
+
+        if not isinstance(value, Time):
+            try:
+                value = self.__class__(value, scale=self.scale, location=self.location)
+            except Exception:
+                try:
+                    value = self.__class__(value, scale=self.scale, format=self.format,
+                                           location=self.location)
+                except Exception:
+                    raise ValueError('cannot convert value to a compatible Time object')
+
+        if self.location != value.location:
```

Comment:
`location` can be an array, in which case this statement would break. Now it is guaranteed that location is either a single element or an array with the correct shape, but the shape may have been broadcast, so one cannot just set an element. So, this needs some trickery.... But it also means in principle a different location is fine. Similarly, if `value.location is None`, I think it might be treated as OK if `self.location is not None` (but arguably only if `self.location` is a single element).